thorlcr/activities/keyedjoin/thkeyedjoin.cpp

```diff
@@ -404,6 +404,30 @@ class CKeyedJoinMaster : public CMasterActivity
                 if (totalIndexParts)
                 {
                     indexMap.map(*this, indexFile, keyHasTlk, getOptBool("allLocalIndexParts"));
+                    if (remoteKeyedLookup && !getOptBool(THOROPT_FORCE_REMOTE_KEYED_LOOKUP))
+                    {
+                        /* heuristic to determin how many nodes in this cluster, have had parts
+                         * mapped for remote lookup.
+                         * If <50%, then turn off remoteKeyedLookup, and process directly
+                         * This scenario can happen, when the index has been written to an overlapping but smaller cluster,
+                         * or if a narrow index has been copied to this cluster (e.g a 1-way hthor index).
+                         */ 
+                        unsigned nodesWithMappedParts = 0;
+                        unsigned groupSize = queryDfsGroup().ordinality();
+                        for (unsigned s=0; s<groupSize; s++)
+                        {
+                            std::vector<unsigned> &parts = indexMap.querySlaveParts(s);
+                            if (parts.size())
+                                nodesWithMappedParts++;
+                        }
+                        if ((nodesWithMappedParts * 2) < groupSize) // i.e. if less than 50% nodes have any parts mapped
```

Comment:
Not worth it, but an option to set this threshold (as a percentage perhaps) might one day be useful ?

Reply:
too many options as is ;) I think on/off is enough in general (can be forced either way already globally or per KJ)
The heuristic here is probably slightly overkill already, i.e. vs just spotting non totally local key and disabling.

Reply:
Fixed typo'